integrations/firehydrant/client.py

```diff
@@ -0,0 +1,451 @@
+            service_metrics[metric_name] //= len(active_incidents_ids)
+
+        return service_metrics
+
+    async def get_single_incident(self, incident_id: str) -> dict[str, Any]:
+        """Retrieve information about a single incident by its unique identifier.
+
+        Args:
+            incident_id (str): The unique identifier of the incident to retrieve.
+
+        Returns:
+            dict[str, Any]: A dictionary containing information about the specified incident.
+
+        """
+        return await self.send_api_request(endpoint=f"incidents/{incident_id}")
+
+    async def compute_incident_analytics_metrics(
+        self, milestones: list[dict[str, Any]]
+    ) -> dict[str, Any]:
+        """
+        Compute incident analytics metrics based on milestone data. Formular retrieved from https://firehydrant.com/docs/managing-incidents/incident-metrics/
+
+        Args:
+            milestones (list[dict[str, Any]]): A list of milestone objects representing incident events.
+
+        Returns:
+            dict[str, Any]: A dictionary containing computed incident analytics metrics.
+
+        """
+
+        incident_times = {
+            "started": None,
+            "acknowledged": None,
+            "mitigated": None,
+            "identified": None,
+            "resolved": None,
+        }
+
+        for event in milestones:
+            incident_type = event["type"]
+            occurred_at = datetime.fromisoformat(event.get("occurred_at", ""))
+
+            if incident_type in incident_times:
+                if incident_times[incident_type] is None:
+                    incident_times[incident_type] = occurred_at  # type: ignore
+            else:
+                continue  # Ignore unknown incident types
+        result: dict[str, Any] = {}
+
+        if incident_times["started"] and incident_times["acknowledged"]:
+            time_to_acknowledge = (
+                incident_times["acknowledged"] - incident_times["started"]
+            ).total_seconds()
+            result["time_to_acknowledge"] = int(time_to_acknowledge / 60)
+
+        if incident_times["started"] and incident_times["identified"]:
+            time_to_identify = (
+                incident_times["identified"] - incident_times["started"]
+            ).total_seconds()
+            result["time_to_identify"] = int(time_to_identify / 60)
+
+        if incident_times["started"] and incident_times["mitigated"]:
+            time_to_acknowledge = (
+                incident_times["mitigated"] - incident_times["started"]
+            ).total_seconds()
+            result["time_to_mitigate"] = int(time_to_acknowledge / 60)
+
+        if incident_times["started"] and incident_times["resolved"]:
+            time_to_resolve = (
+                incident_times["resolved"] - incident_times["started"]
+            ).total_seconds()
+            result["time_to_resolve"] = int(time_to_resolve / 60)
+
+        return result
```

Comment:
im not sure this is needed
correct me if im wrong but u can just pass the milesstones and have some calculation properties over it